keanu-python/keanu/vertex/vertex_helpers.py

```diff
@@ -0,0 +1,54 @@
+from typing import Callable, Dict, Any, Collection
+from .base import (Vertex, cast_tensor_arg_to_double, cast_tensor_arg_to_integer, cast_tensor_arg_to_bool,
+                   vertex_constructor_param_types)
+from keanu.tensor import Tensor
+from keanu.context import KeanuContext
+from keanu.infer_type import infer_type_and_execute
+from keanu.vartypes import tensor_arg_types
+from py4j.java_collections import JavaArray
+
+k = KeanuContext()
+
+
+def do_vertex_cast(vertex_ctor: Callable, value: vertex_constructor_param_types) -> Vertex:
+    return value if isinstance(value, Vertex) else vertex_ctor(value)
+
+
+def do_generic_vertex_cast(ctors: Dict[type, Callable], value: vertex_constructor_param_types) -> Vertex:
```

Comment:
This method name makes it seem like we are specifically casting `GenericVertex` (which I suppose `Vertex` kind of is...? haha)

Reply:
Yeah - one of the little warts on our class hierarchy.  I'll rename.